File: app/src/main/java/com/constraint/vagabond/data/RecreationalArea.java
package com.constraint.vagabond.data;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class RecreationalArea {

  @SerializedName("RecAreaID")
  @Expose
  public final String recAreaID;

  @SerializedName("RecAreaName")
  @Expose
  public final String recAreaName;

  @SerializedName("RecAreaDescription")
  @Expose
  public final String recAreaDescription;

  @SerializedName("RecAreaDirections")
  @Expose
  public final String recAreaDirections;

  @SerializedName("RecAreaPhone")
  @Expose
  public final String recAreaPhone;

  @SerializedName("RecAreaEmail")
  @Expose
  public final String recAreaEmail;

  @SerializedName("MEDIA")
  @Expose
  public final List<RecAreaMedia> recAreaMediaList;

  public RecreationalArea(
      String recAreaID,
      String recAreaName,
      String recAreaDescription,
      String recAreaDirections,
      String recAreaPhone,
      String recAreaEmail,
      List<RecAreaMedia> recAreaMediaList) {
    this.recAreaID = recAreaID;
    this.recAreaName = recAreaName;
    this.recAreaDescription = recAreaDescription;
    this.recAreaDirections = recAreaDirections;
    this.recAreaPhone = recAreaPhone;
    this.recAreaEmail = recAreaEmail;
    this.recAreaMediaList = recAreaMediaList;
  }
}

File: app/src/main/java/com/sierra/vagabond/data/entities/RecreationalArea.kt
package com.sierra.vagabond.data.entities

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.sierra.vagabond.data.local.DataConverter
import com.sierra.vagabond.utils.*
import kotlinx.android.parcel.Parcelize

@Entity(tableName = TABLE_NAME)
@Parcelize
data class RecreationalArea(@field:PrimaryKey @field:SerializedName(REC_AREA_ID) @field:Expose val recAreaID: String,
                            @field:SerializedName(REC_AREA_NAME) @field:Expose val recAreaName: String,
                            @field:SerializedName(REC_AREA_DESC) @field:Expose val recAreaDescription: String,
                            @field:SerializedName(REC_AREA_DIRS) @field:Expose val recAreaDirections: String,
                            @field:SerializedName(REC_AREA_PHONE) @field:Expose val recAreaPhone: String,
                            @field:SerializedName(REC_AREA_EMAIL) @field:Expose val recAreaEmail: String,
                            @field:SerializedName(REC_AREA_MEDIA) @field:Expose @field:TypeConverters(DataConverter::class) val recAreaMediaList: List<RecAreaMedia>,
                            @field:SerializedName(REC_AREA_FACILITIES) @field:Expose @field:TypeConverters(DataConverter::class) val recAreaFacilities: List<RecAreaFacilities>) : Parcelable
